src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -402,36 +402,41 @@ const NodeResources &ClusterResourceScheduler::GetLocalNodeResources() const {
   return node_it->second.GetLocalView();
 }
 
-int64_t ClusterResourceScheduler::NumNodes() { return nodes_.size(); }
+int64_t ClusterResourceScheduler::NumNodes() const { return nodes_.size(); }
 
-void ClusterResourceScheduler::AddLocalResource(const std::string &resource_name,
-                                                double resource_total) {
-  string_to_int_map_.Insert(resource_name);
-  int64_t resource_id = string_to_int_map_.Get(resource_name);
+const StringIdMap ClusterResourceScheduler::GetStringIdMap() const {
```

Comment:
This too. Return the reference instead. 

```suggestion
const StringIdMap &ClusterResourceScheduler::GetStringIdMap() const {
```